compiler/noirc_evaluator/src/ssa/opt/flatten_cfg.rs

```diff
@@ -698,12 +748,12 @@ impl<'f> Context<'f> {
                     Instruction::Constrain(lhs, rhs, message)
                 }
                 Instruction::Store { address, value } => {
-                    // If this instruction immediately follows an allocate, and stores to that
-                    // address there is no previous value to load and we don't need a merge anyway.
+                    // There is no side effect for 'Store' instructions on arrays allocated in the current branch, because
+                    // these arrays do not exist in other branches.
```

Comment:
I think the old comment may have been more accurate here. Since this check isn't for handling side-effects but rather for merging the previous value with the current value, which isn't possible as the comment notes if there was no previous value. Additionally it can be any value, not just arrays.

Reply:
Done